### PR TITLE
[#140] Raise Dependencies

### PR DESCRIPTION
**Pull request by notmyown**
- Update Groovy to 3.0.23
- Update JUnit to 5.11.3 and added groovy-test-junit5:3.0.23
- Update Mockito to 4.5.1
- Update Hamcrest to 3.0
- Update Maven to 3.9.9
- maven-compiler-plugin to 3.13.0

### CODE REVIEW

**Comment by schnatterer** [comment]
Sorry for chiming in unasked. 
Please don't upgrade Groovy version.
AFAIK Groovy in Jenkins is still on 2.x (see screenshot bellow). 

Until before this PR, leaving the version on 2 protected us from using groovy code that would later fail on Jenkins because of lacking syntactic support.

It would be great, though, to use latest Groovy 4 features, e.g. in test and for latest security fixes.
Maybe there is way to configure the groovy compiler to check groovy 2 compatibility mode, similar to Javas source level?

image

File: test/com/cloudogu/ces/cesbuildlib/K3dTest.groovy [hunk 1]
```diff
@@ -9,7 +11,7 @@ import static org.mockito.ArgumentMatchers.anyString
 import static org.mockito.Mockito.mock
 import static org.mockito.Mockito.when
 
-class K3dTest extends GroovyTestCase {
+class K3dTest {
```
Comment (ppxl) [inline test/com/cloudogu/ces/cesbuildlib/K3dTest.groovy:14]: Finally arriving in year 2010 with tests that no longer extend other testcases :laughing:

**Review by ppxl — APPROVED** [review]
Great work, thank you. LGTM

**Comment by ppxl** [comment]
> Sorry for chiming in unasked. Please don't upgrade Groovy version. AFAIK Groovy in Jenkins is still on 2.x (see screenshot bellow).
> 
> Until before this PR, leaving the version on 2 protected us from using groovy code that would later fail on Jenkins because of lacking syntactic support.
> 
> It would be great, though, to use latest Groovy 4 features, e.g. in test and for latest security fixes. Maybe there is way to configure the groovy compiler to check groovy 2 compatibility mode, similar to Javas source level?
> 
> image

Oh that's an important note which I didn't had on my 2024 card: "Tooling still on Groovy 2.x" :disappointed: 

Thanks for the info :clap:

**Comment by notmyown** [comment]
PR obsolet due to other Branch - fixing the Groovy 3 - Jenkins issue

**Pull request closed without merging**